backend/src/api/artist_routes.py

```diff
@@ -0,0 +1,149 @@
+from flask import Blueprint, request
+from flask_login import current_user  # pyright: ignore
+from ..models import User, db
+from typing import cast, Union, Tuple
+from datetime import datetime
+from ..backend_api import (
+    Artist,
+    PostArtist, 
+    ApiError, 
+)
+
+artist_routes = Blueprint("artists", __name__, url_prefix="/api/artists")
+
+@artist_routes.get("/<int:artist_id>")
+def get_artist(artist_id: int) -> Union[Artist, Tuple[ApiError, int]]:
+    """
+    Get an artist's details by ID.
+    Returns (ApiError, 404) if artist not found or if user exists but isn't an artist.
+    """
+    artist = db.session.query(User).get(artist_id)
+    
+    if not artist:
+        return (ApiError(
+            message="Artist not found",
+            errors={"artist_id": f"No user found with id {artist_id}"}
+        ), 404)
+    
+    if not artist.stage_name:
+        return (ApiError(
+            message="Not an artist",
+            errors={"artist_id": f"User {artist_id} is not an artist"}
+        ), 404)
+    
+    # Build artist response
+    result: Artist = {
+        "id": artist.id,
+        "stage_name": artist.stage_name
+    }
+    
+    # Add optional fields if they exist
+    if artist.profile_image:
+        result["profile_image"] = artist.profile_image
+    if artist.first_release:
+        result["first_release"] = str(artist.first_release)
+    if artist.biography:
+        result["biography"] = artist.biography
+    if artist.location:
+        result["location"] = artist.location
+    if artist.homepage:
+        result["homepage"] = artist.homepage
+        
+    return result
+
+@artist_routes.post("")
+def post_artist() -> Union[PostArtist, Tuple[ApiError, int]]:
```

Comment:
because we are using py3.12, this could also be written as 
```suggestion
def post_artist() -> PostArtist | tuple[ApiError, int]:
```
and as of the latest changes to main, it would be `ApiErrorResponse` instead of the tuple